adding Eset SMC parser

tkubica12 opened the pull request:
Fixes #

## Proposed Changes

  -
  -
  -

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/Azure-Sentinel?pullRequest=476) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/Azure-Sentinel?pullRequest=476) before we can accept your contribution.<br/><br/>:x: tkubica12 [sign now](https://cla.opensource.microsoft.com/Azure/Azure-Sentinel?pullRequest=476)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/Azure-Sentinel?pullRequest=476) it.</sub>

tkubica12 commented:
Thanks for help! I have rewriten instructions into dataconnector format, please review.

preetikr commented on DataConnectors/esetSmc.json:
```diff
@@ -0,0 +1,117 @@
+    "id": "EsetSMC",
+    "title": "Eset Security Management Center",
+    "publisher": "Eset",
+    "descriptionMarkdown": "Connector to bring Eset SMC data into Sentnel using custom table for easy quering. Data is sent from Eset SMB in JSON format to syslog connector VM and FluentD parsing is used so data are sent structured to Sentinel.",
```
Please ensure it's Azure Sentinel everywhere. Here there's a typo as well. Is it possible to provide details on what type of Eset data is available instead of providing info on the how the data is sent? The how part is covered in the configuration section.

preetikr commented on DataConnectors/esetSmc.json:
```diff
@@ -0,0 +1,117 @@
+    "publisher": "Eset",
+    "descriptionMarkdown": "Connector to bring Eset SMC data into Sentnel using custom table for easy quering. Data is sent from Eset SMB in JSON format to syslog connector VM and FluentD parsing is used so data are sent structured to Sentinel.",
+    "graphQueries": [
+        {
```
Looks like the data lands in custom logs. Please update this to insert in the DATATYPE_NAME which is eset_CL for both legend and baseQuery.

preetikr commented on DataConnectors/esetSmc.json:
```diff
@@ -0,0 +1,117 @@
+    ],
+    "dataTypes": [
+        {
+            "name": "DATATYPE_NAME",
```
Replace DATATYPE_NAME with the right name (eset_CL).

preetikr commented on DataConnectors/esetSmc.json:
````diff
@@ -0,0 +1,117 @@
+      },
+      {
+          "title": "2. Configure the logs to be collected",
+          "description": "Configure rsyslog to accept logs from your Eset SMC IP address.\n\n```\nsudo -i\r\n\r\n# Set ESET SMC source IP address\r\nexport ESETIP=1.2.3.4\r\n\r\n# Create rsyslog configuration file\r\ncat > \/etc\/rsyslog.d\/80-remote.conf << EOF\r\n\\$ModLoad imudp\r\n\\$UDPServerRun 514\r\n\\$ModLoad imtcp\r\n\\$InputTCPServerRun 514\r\n\\$AllowedSender TCP, 127.0.0.1, $ESETIP\r ...
````
Please write up all commands in the instructions on a separate line with "Copyable label" also associated with those. Refer to Syslog template as an example of how you can have a copyable label that will enable customers to copy the command and run it in their console easily without copying extra / less and run into issues.

preetikr commented on DataConnectors/esetSmc.json:
````diff
@@ -0,0 +1,117 @@
+      },
+      {
+          "title": "3. Configure OMS agent to pass EsetSMC data in API format",
+          "description": "In order to easily recognize Eset data we will push it to separated table. To make it simple we will just modify match oms. section to send data as API objects by changing type to out_oms_api. Modify file on \/etc\/opt\/microsoft\/omsagent\/(REPLACEyourworkspaceid)\/conf\/omsagent.conf. Full match oms. section looks like this:\r\n\r\n```\r\n<match oms.** docker.**>\r\n   ...
````
Why are we doing this? Why not push to Syslog table and just snap to the Syslog template format? Why converting Syslog to REST API? Also if you push into Syslog you can use the conversion to REST API as a parser/Kusto Function to well format and enable customers to work with the data.

tkubica12 commented:
Fixed issues and enhanced connector based on feedback. Also added workbook and detections. Please check and let me know whether further modifications are needed. Thanks!
